### ISSUE TITLE
In the new Visual Studio 2015 RC when calling services.AddReact() - The type 'IServiceCollection' is defined in an assembly that is not referenced

### ISSUE THREAD

**Issue by chrisjhanna** [reactjs/React.NET#125]
Compile time error in Startup.cs on line services.AddReact():

"The type 'IServiceCollection' is defined in an assembly that is not referenced"

This was in a project that we had been working on in Visual Studio 2015 CTP 5. Everything was working as normal until we upgraded the project to Visual Studio 2015 RC. All the other services are working fine after the upgrade. There seems to have been quite a few API changes in RC so something may have been moved that is affecting it.

**Comment by st1pps** [comment]
The current version of react.js is targeting asp.net 5 beta-3, while VS 2015 RC automatically references beta-4. It seems that there are quite a few changes between the two beta versions.

**Comment by Daniel15** [comment]
That's to be expected when dealing with prerelease software :)

I'll try to get some time to update it based on the beta4 changes.

**Comment by stefanwuthrich** [comment]
yes please. i try'd to upgrade to beta4, but to complex for me :-(

**Comment by st1pps** [comment]
After updating all references and changing the framework from ASPNET50 to DNX 4.5.1, I just had to edit JsxFileSystem.cs to make it compile and pass all test: 
```csharp
private class JsxFileInfo : IFileInfo
		{
			private readonly IJsxTransformer _jsxTransformer;
			private readonly IFileInfo _fileInfo;
			private readonly Lazy<byte[]> _content;

			public JsxFileInfo(IJsxTransformer jsxTransformer, IFileInfo fileInfo)
			{
				_jsxTransformer = jsxTransformer;
				_fileInfo = fileInfo;

				_content = new Lazy<byte[]>(
					() => Encoding.UTF8.GetBytes(_jsxTransformer.TransformJsxFile(fileInfo.PhysicalPath))
				);
			}

			public Stream CreateReadStream()
			{
				return new MemoryStream(_content.Value);
			}

			public long Length
			{
				get { return _content.Value.Length; }
			}

			public string PhysicalPath
			{
				get { return null; }
			}

			public string Name
			{
				get { return _fileInfo.Name; }
			}

			public bool IsDirectory
			{
				get { return _fileInfo.IsDirectory; }
			}

#if OWIN
			public DateTime LastModified
			{
				get { return _fileInfo.LastModified; }
			}
#else
			public DateTimeOffset LastModified
			{
				get { return _fileInfo.LastModified; }
			}

			public void WriteContent(byte[] content)
			{
                File.WriteAllBytes(_fileInfo.PhysicalPath, content);
			}

			public void Delete()
			{
                File.Delete(_fileInfo.PhysicalPath);
			}

			public bool Exists
			{
				get { return _fileInfo.Exists; }
			}

			public bool IsReadOnly
			{
               
				get { return File.GetAttributes(_fileInfo.PhysicalPath).HasFlag(FileAttributes.ReadOnly); }
			}
#endif
		}
```

In the React.Sample.MVC6 project the app.UseRequestServices(); statement had to be removed, as the method is not longer existing. As far as I can tell, everything is working fine.

**Comment by stefanwuthrich** [comment]
@st1pps Thank you so much. You saved my day. works perfect ;-)

**Comment by Daniel15** [comment]
Just released 1.5.2 which updates the ASP.NET dependency to beta 4. Thanks @st1pps!

**Comment by chrisjhanna** [comment]
Awesome, thanks for this.
